com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/ZigBeeDongleEzsp.java

```diff
@@ -396,6 +412,19 @@ public ZigBeeStatus startup(boolean reinitialize) {
                 : ZigBeeStatus.BAD_RESPONSE;
     }
 
+    private void scheduleNetworkStatePolling() {
+        if (pollingTimer != null) {
+            pollingTimer.cancel(true);
+        }
+
+        pollingTimer = pollingScheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
```

Comment:
I'm wondering if we shouldn't be a little smarter with this to reduce traffic and only poll if there has been no activity? Maybe it also doesn't really matter, but I'm just a little hesitant to pump frames into the ```FrameHandler``` if it is already processing a lot of data (as might happen during initialisation). I guess overhead is probably low enough that it's ok...

Reply:
I agree, that making it smart would be a nice idea. However, I cannot ask the `frameHandler` about its current queue size, its not in ins API. If we would extend the API I could simple ask it whether there are already frames in there (which will also run in a retry timeout anyway) and not queue another one.

Otherwise, I think one additional frame per second doesn't hurt too much.